lib/gooddata/lcm/actions/synchronize_new_segments.rb

```diff
@@ -28,7 +28,8 @@ def call(params)
 
           domain_name = params.organization || params.domain
           domain = client.domain(domain_name) || fail("Invalid domain name specified - #{domain_name}")
-          domain_segments = domain.segments
+          data_product = params.data_product
```

Comment:
Refers to 'params' more than self (maybe move it to another class?) (FeatureEnvy)